tests/unit/base/Plugin_unittest.cc

```diff
@@ -0,0 +1,31 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2016 Sanuj Sharma
+ */
+
+#include <../tests/unit/base/MockBaseClass.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+
+TEST(Plugin, load_close)
+{
+    auto library = loadLibrary("./libplugin.so");
```

Comment:
@karlnapf @lisitsyn `createX()` calls `loadLibrary()` and other functions to return an object of `X` to the end user.